src/test/java/api/exercise/Exercise2.java

```diff
@@ -0,0 +1,127 @@
+package api.exercise;
+
+import org.junit.Test;
+
+import java.util.function.BinaryOperator;
+
+import static org.junit.Assert.*;
+
+public class Exercise2 {
+
+    @Test
+    public void sequentialPrefixTestSum() {
+        Integer[] original = {1, 2, 3, 4, 5, 6, 7, 8};
+
+        Integer[] result = sequentialPrefix(original, Integer::sum);
+
+        assertNotSame(original, result);
+        assertArrayEquals(new Integer[]{1, 3, 6, 10, 15, 21, 28, 36}, result);
+    }
+
+    @Test
+    public void sequentialPrefixTestMult() {
+        Integer[] original = {1, 2, 3, 4, 5, 6, 7, 8};
+
+        Integer[] result = sequentialPrefix(original, (left, right) -> left * right);
+
+        assertNotSame(original, result);
+        assertArrayEquals(new Integer[]{1, 2, 6, 24, 120, 720, 5_040, 40_320}, result);
+    }
+
+    @Test
+    public void sequentialPrefixTestLiningByMax() {
+        Integer[] original = {1, 2, 3, 2, 1, 5, 7, 5};
+
+        Integer[] result = sequentialPrefix(original, Integer::max);
+
+        assertNotSame(original, result);
+        assertArrayEquals(new Integer[]{1, 2, 3, 3, 3, 5, 7, 7}, result);
+    }
+
+    /**
+     * Выполняет операцию сканирования в однопоточном режиме.
+     * Не модифицирует исходный набор данных.
+     *
+     * @param source   Массив исходных элементов.
+     * @param operator Оператор сканирования.
+     * @return Результат сканирования.
+     * @see <a href="https://habr.com/company/epam_systems/blog/247805">Сканирование</a>
+     */
+    private static <T> T[] sequentialPrefix(T[] source, BinaryOperator<T> operator) {
```

Comment:
You should implement this method using the scan algorithm from the habr article (follow the link in javadoc).